### DISCUSSION TITLE
C3D creation tutorial

### DISCUSSION THREAD

**Discussion by BiomedicLJZ** [kineticstoolkit/kineticstoolkit#142] [2. Development: Ideas, new features, doc, etc.]
Hi, as biomedical engineer one huge part of my work include biomechanics and sometimes I don't have the access to C3D files so I need to create some of but it have been a great challenge with poor documentation at least what I've found

**Comment by felixchenier** [comment]
Hello @BiomedicLJZ 

To create a c3d using Kinetics Toolkit, use [ktk.write_c3d](https://kineticstoolkit.uqam.ca/doc/api/ktk.write_c3d.html)*. It takes a ktk.TimeSeries as an input. The TimeSeries must be formatted as this sample [here](https://kineticstoolkit.uqam.ca/doc/kinematics_load_visualize.html), where each element of the TimeSeries.data field are Nx4 arrays. You can see this [introduction to TimeSeries](https://kineticstoolkit.uqam.ca/doc/timeseries.html) for more information on how to work with ktk.TimeSeries.

Does it solve your need?

Now, for your suggestion, is what you're thinking about would be some kind of cookbook, where we have different short codes (recipes) for this or that operation? Like:
- How to create a TimeSeries from a C3D
- How to write a C3D from a TimeSeries
- etc.

I think I can see the usefulness of such recipes, but I have difficulty to see where to fit those between full-blown tutorials (the Kinetics Toolkit [manual](https://kineticstoolkit.uqam.ca/doc/ktk_manual.html)) and the function descriptions (the [API](https://kineticstoolkit.uqam.ca/doc/api_reference.html)). Would improving the docstring of ktk.write_c3d be sufficient in your opinion?

*Note that writing analog data is still in development for now.

**Reply by BiomedicLJZ** [reply]
Thank you for your quick response 
In my opinion that kind of content would fit better in the docstring so a little bit of improvement there should be sufficient. Thanks again for everything
